### PR TITLE
Update `Variable.set` return value + docs

### PR DESCRIPTION
**Pull request by jakekaplan**
This PR updates `Variable.set` to return the newly set `value` by default. If you pass `as_object` it will return the entire `Variable` object. This makes it a consistent interface with the `get` method.

This pr also updates the docs for the new `get`, `set` and `unset` interfaces. It also take into account the changes to `@sync_compatible` for when these methods can be used synchronously vs asynchronously.

### CODE REVIEW

File: docs/guides/variables.md [hunk 1]
```diff
@@ -44,51 +44,87 @@ You can create and delete variables through the REST API. You can also set and g
...
 ## Accessing variables
 
 In addition to the UI and API, you can reference variables in code and in certain Prefect configuration files.
 
 ### In Python code
 
-You can access any variable through the Python SDK with the `Variable.get()` method. If you attempt to reference a variable that does not exist, the method will return `None`. You can create variables through the Python SDK with the `Variable.set()` method. Note that if a variable of the same name exists, you'll need to pass `overwrite=True`.
+You can interact with Variables via the Python SDK utilizing the `get`, `set`, and `unset` methods. 
+
+Note that these methods are invoked either synchronously or asynchronously, depending on the context in which they are called
```
Comment (discdiver) [inline docs/guides/variables.md:61]:
```suggestion
Note that these methods are invoked either synchronously or asynchronously, depending on the context in which they are called. 
```

File: docs/guides/variables.md [hunk 2]
```diff
@@ -44,51 +44,87 @@ You can create and delete variables through the REST API. You can also set and g
...
+while inside of a `flow` or `task`. Outside of a `flow` or `task`, these methods are always asynchronous.
```
Comment (discdiver) [inline docs/guides/variables.md:62]:
```suggestion
Outside of a `flow` or `task`, these methods are always asynchronous.
```

File: docs/guides/variables.md [hunk 3]
```diff
@@ -44,51 +44,87 @@ You can create and delete variables through the REST API. You can also set and g
...
+    await Variable.unset("the_answer")
+    
+    # get a variable that doesn't exist
+    answer = await Variable.get("not_the_answer")
+    print(answer) # None
+    
+    # get a variable that doesn't exist with a default
+    answer = await Variable.get("not_the_answer", default="42")
+    print(answer) # 42
 ```
```
Comment (discdiver) [inline docs/guides/variables.md:95]: I think this is a good spot to note what's happening here with: "While inside of a `flow` or `task`...

File: docs/guides/variables.md [hunk 4]
```diff
@@ -44,51 +44,87 @@ You can create and delete variables through the REST API. You can also set and g
 
 ### Through the CLI
 
-List, inspect, and delete variables in the CLI with the `prefect variable ls`, `prefect variable inspect <name>`, and `prefect variable delete <name>` commands, respectively.
-
+- `prefect variable set` creates or updates a variable.
+- `prefect variable get` retrieves a variable's value.
+- `prefect variable unset` deletes a variable.
+- `prefect variable ls` lists all variables.
+- `prefect variable inspect` shows a variable's details.
```
Comment (billpalombi) [inline docs/guides/variables.md:51]: What does "details" mean here?

**Review by billpalombi — COMMENTED** [review]
Couple nits

File: docs/guides/variables.md [hunk 5]
```diff
@@ -44,51 +44,87 @@ You can create and delete variables through the REST API. You can also set and g
...
+    # get a variable that doesn't exist
+    answer = await Variable.get("not_the_answer")
+    print(answer) # None
+    
+    # get a variable that doesn't exist with a default
+    answer = await Variable.get("not_the_answer", default="42")
+    print(answer) # 42
+```
 
+While inside a `flow` or `task`, they should be invoked synchronously or asynchronously depending on if the current
```
Comment (discdiver) [inline docs/guides/variables.md]:
```suggestion
While inside a `flow` or `task`, they should be invoked synchronously or asynchronously depending on whether the current context is synchronous or asynchronous.
```

**Pull request merged**